core/src/mender-zephyr-image-update-module.c

```diff
@@ -29,6 +29,8 @@
  */
 static void *mcu_boot_flash_handle = NULL;
 
+static bool artifact_had_payload;
```

Comment:
(in the context of my previous comment)

The variable name reads strange to me,

what about `mcu_boot_image_flashed` or something like that? The prefix `mcu_boot`for consistency with the above and the name `image_flashed` to relate it to the action (flashing) rather than the trigger or source (the payload)